EIPS/eip-7668.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 7668
+title: Add Max Epoch Churn Limit
+description: Modify the churn limit function to upper bound it to a max value
+author: dapplion (@dapplion)
+discussions-to: https://ethereum-magicians.org/t/eip-7668-add-max-epoch-churn-limit/15709
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-07
+---
+
+## Abstract
+
+Limit Beacon chain growth from exponential to linear to mitigate the risks of very high levels of total ETH supply staked.
+
+## Motivation
+
+With a full deposit queue, the share of ETH supply staked will reach 50% by May 2024, 75% by September 2024, and 100% by December 2024. While yields decrease as the validator set size increases, the APY at 100% ETH supply staked is 1.6% for consensus rewards only excluding MEV. This yield can be sufficient to attract capital such that the equilibrium point of the validator set size is close to its maximum possible.
+
+Reaching very high levels of total ETH supply staked can have negative externalities. Liquid staking derivatives (LSDs) are likely to absorb most of the staking demand, jeopardizing the economic guarantees of the Beacon chain. If LSD tokens overtake the native token in liquidity, they permeate systemic risks into most Ethereum applications.
+
+It's unclear if the Ethereum network security benefits from having so much stake. After some upper bound, attacks become so economically unfeasible that additional stake does not increase the network's safety. In that case, the network will be overpaying for security. Note that Beacon chain rewards are dictated by an opinionated function, chosen before its launch in 2020. Much has changed since then, including the Beacon chain achieving unprecedented success. Beacon chain rewards should be adjusted to match today's reality, and stop incentiving actors into increasing the risks listed above at no benefit to the network's security.
+
+If one agrees that Beacon chain rewards should be adjusted down, it's important to forecast the social cost of doing so. The more ETH is staked, the more actors may depend on today's reward regime and oppose the change. Also, the network could reach some catastrophic tipping point before it can agree to a new rewards function.
+
+This proposal accepts the complexities of changing the rewards curve and is meant only to slow down growth. It wants to mitigate the negative externalities of very high levels of total ETH supply staked before a proper solution is implemented.
+
+## Specification
+
+### Constants
+
+| Name | Value |
+| ---- | ----- |
+| `MAX_PER_EPOCH_CHURN_LIMIT` | 12 |
```

Comment:
I know this is the current value, but have you considered reducing it to a more modest number?
Why 12 and not, say, 8 or 6?

Reply:
Is there consideration in only applying this to activations and not exits?

Reply:
> Is there consideration in only applying this to activations and not exits?

At the first glance such imbalance doesn't affect security properties, but it might cause the need of additional analysis before the proposal can be accepted. Considering short time frames we might not want to impose any additional complexity to make this proposal pass through the governance process. Thus, capping both queues is preferable, IMHO

Reply:
The lower the number the higher delta there will be in queue capacity at the fork epoch. A lower number increases the "unfairness" argument, but it increases the effectiveness of this EIP. I pick 12 without strong reasoning, just because it yields good results without resulting in a dramatic drop in churn.

Reply:
One of the potential arguments in favour of `12` is that this is likely the limit when the Deneb fork happens on the mainnet assuming the same market conditions, so the decrease of the churn limit is unlikely to happen